src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+package org.opentripplanner.routing.impl;
+
+import org.opentripplanner.model.Route;
+import org.opentripplanner.routing.core.Fare;
+import org.opentripplanner.routing.core.FareRuleSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Currency;
+import java.util.List;
+
+/**
+ * Calculate Orca discount fares based on the fare type and the agencies traversed within a trip.
+ */
+public class OrcaFareServiceImpl extends DefaultFareServiceImpl {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OrcaFareServiceImpl.class);
+
+    public enum RideType {
+        COMM_TRANS_LOCAL_SWIFT,
+        COMM_TRANS_COMMUTER_EXPRESS,
+        EVERETT_TRANSIT,
+        KC_WATER_TAXI_VASHON_ISLAND,
+        KC_WATER_TAXI_WEST_SEATTLE,
+        KC_METRO,
+        KITSAP_TRANSIT,
+        PIERCE_COUNTY_TRANSIT,
+        SEATTLE_STREET_CAR,
+        SOUND_TRANSIT,
+        WASHINGTON_STATE_FERRIES
+    }
+
+    // If set to true, the test ride price is used instead of the actual agency cash fare.
+    public boolean IS_TEST;
+    public static final float DEFAULT_TEST_RIDE_PRICE = 3.49f;
+
+    public OrcaFareServiceImpl(Collection<FareRuleSet> regularFareRules) {
+        addFareRules(Fare.FareType.regular, regularFareRules);
+        addFareRules(Fare.FareType.senior, regularFareRules);
+        addFareRules(Fare.FareType.youth, regularFareRules);
+        addFareRules(Fare.FareType.orcaRegular, regularFareRules);
+        addFareRules(Fare.FareType.orcaYouth, regularFareRules);
+        addFareRules(Fare.FareType.orcaLift, regularFareRules);
+        addFareRules(Fare.FareType.orcaSenior, regularFareRules);
+    }
+
+    /**
+     * Classify the ride type based on the route information provided. In most cases the agency name is sufficient. In
+     * some cases the route description and short name are needed to define inner agency ride types.
+     */
+    private static RideType classify(Route routeData) {
+        if ("29".equals(routeData.getAgency().getId())) {
```

Comment:
Make an enum for all these agency IDs. See [here for a good example](https://stackoverflow.com/a/3978690/269834).